package/tests/test_models/test_graph/test_graph_nodes.py

```diff
@@ -95,6 +95,40 @@ def test_create_task_node(self, namespace, expected_modular_pipelines):
         assert task_node.pipelines == set()
         assert task_node.modular_pipelines == expected_modular_pipelines
 
+    
+    @pytest.mark.parametrize(
+        "namespace,expected_modular_pipelines",
+        [
+            (None, []),
+            (
+                "uk.data_science.model_training",
+                [
+                    "uk",
+                    "uk.data_science",
+                    "uk.data_science.model_training",
+                ],
+            ),
+        ],
+    )
+    def test_create_task_node_no_given_name(self, namespace, expected_modular_pipelines):
+        kedro_node = node(
+            identity,
+            inputs="x",
+            outputs="y",
+            tags={"tag"},
+            namespace=namespace,
+        )
+        task_node = GraphNode.create_task_node(kedro_node)
+        assert isinstance(task_node, TaskNode)
+        assert task_node.kedro_obj is kedro_node
+        assert task_node.id == GraphNode._hash(str(kedro_node))
+        assert task_node.name == "Identity"
+        assert task_node.full_name == "identity"
```

Comment:
You just need this one assert in this test. No need for any other assert. You also don't need the modular pipeline parametrize here. It's not relevant to what we want to test with this test.